src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -709,6 +705,10 @@ public function link($method = 'get')
 
         $node = $this->getNode(0);
 
+        if (!$node instanceof \DOMElement) {
+            throw new \InvalidArgumentException(sprintf('The current node list should contain only DOMElement instances, "%s" found.', get_class($node)));
```

Comment:
Imo the message is a bit misleading as you only check the type of the first element.